File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        maps = {}

        for i in range(len(nums)):
            diff = target - nums[i]

            if nums[i] in maps:
                return [maps[nums[i]], i]

            maps[diff] = i

        

File: Data Structures & Algorithms/search-insert-position/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    searchInsert(nums, target) {
        let l = 0;
        let r = nums.length - 1;

        while (l <= r) {
            let m = Math.floor((r + l) / 2);
            const cur = nums[m];

            if (cur === target) {
                return m;
            } else if (cur > target) {
                r = m - 1;
            } else {
                l = m + 1;
            }
        }

        return l;
    }
}
